addons/web/static/src/js/views/basic/basic_model.js

```diff
@@ -1439,6 +1439,11 @@ var BasicModel = AbstractModel.extend({
                 record._rawChanges[name] = val;
                 return;
             }
+            if (record._rawChanges[name]) {
+                // if previous _rawChanges exists, clear them since the field is now knwon
+                // and restoring outdated onchange over posterious change is wrong
+                delete record._rawChanges[name];
```

Comment:
Are you sure that rawChanges have been applied before deleting them? Or should they not be applied?

Reply:
as far as I can tell when we get the onchange of unloaded data we get either existing data from the database or data from real `@api.onchange` methods.

and when the record is loaded we pass by:

https://github.com/odoo/odoo/blob/9148d86f35fe99c0af21ade9d497e8534510aac6/addons/web/static/src/js/views/basic/basic_view.js#L67
https://github.com/odoo/odoo/blob/9148d86f35fe99c0af21ade9d497e8534510aac6/addons/web/static/src/js/views/basic/basic_view.js#L124-L130
https://github.com/odoo/odoo/blob/9148d86f35fe99c0af21ade9d497e8534510aac6/addons/web/static/src/js/views/basic/basic_model.js#L293-L295

so if we had unloaded _rawChanges over the field, the first time the field is loaded we will pass in _applyOnChange applying them (and with the fix removing them so they will only be applied one time when the field was loaded)